.github/scripts/generate-scan-policies.js

```diff
@@ -0,0 +1,77 @@
+// This is a ZAP standalone script - it will only run in ZAP.
+// It generates the scan policies for https://github.com/zaproxy/zap-extensions/tree/main/addOns/scanpolicies etc
+// The policies are created after starting a ZAP weekly release with the '-addoninstall ascanrulesAlpha' option.
+
+var FileWriter = Java.type("java.io.FileWriter");
+var PrintWriter = Java.type("java.io.PrintWriter");
+var PolicyTag = Java.type("org.zaproxy.addon.commonlib.PolicyTag");
+var activeScanScript = Java.type(
+  "org.zaproxy.zap.extension.scripts.scanrules.ScriptsActiveScanner"
+);
+var StringEscapeUtils = Java.type("org.apache.commons.text.StringEscapeUtils");
+
+var extAscan = control
+  .getExtensionLoader()
+  .getExtension(org.zaproxy.zap.extension.ascan.ExtensionActiveScan.NAME);
+
+var plugins = extAscan
+  .getPolicyManager()
+  .getDefaultScanPolicy()
+  .getPluginFactory()
+  .getAllPlugin()
+  .toArray()
+  .sort(function (a, b) {
+    return a.getId() - b.getId();
+  });
+
+var INDENT = "    ";
+for (var idx = 0; idx < PolicyTag.values().length; idx++) {
+  var currentTag = PolicyTag.values()[idx];
+  var policyFilePath =
+    "/zap/wrk/zap-extensions/addOns/XXXXX/src/main/zapHomeFiles/policies/".replace(
+      "XXXXX",
+      currentTag.getAddonId()
+    ) + currentTag.getFileName();
+  print(policyFilePath);
+  // Create the policy
+  var fw = new FileWriter(policyFilePath);
+  var pw = new PrintWriter(fw);
+  pw.println('<?xml version="1.0" encoding="UTF-8" standalone="no"?>');
+  pw.println("<configuration>");
+  pw.println(
+    INDENT +
+      "<policy>" +
+      StringEscapeUtils.escapeXml11(currentTag.getPolicyName()) +
+      "</policy>"
+  );
+  pw.println(INDENT + "<scanner>");
+  pw.println(INDENT.repeat(2) + "<level>OFF</level>");
+  pw.println(INDENT.repeat(2) + "<strength>MEDIUM</strength>");
+  pw.println(INDENT + "</scanner>");
+  pw.println(INDENT + "<plugins>");
+
+  for (var i = 0; i < plugins.length; i++) {
+    try {
+      if (
+        plugins[i].getAlertTags() != null &&
+        plugins[i].getAlertTags().keySet().contains(currentTag.getTag())
+      ) {
+        pw.println(INDENT.repeat(2) + "<p" + plugins[i].getId() + ">");
+        pw.println(
+          INDENT.repeat(3) +
+            "<name>" +
+            StringEscapeUtils.escapeXml11(plugins[i].getName()) +
+            "</name>"
+        );
+        pw.println(INDENT.repeat(3) + "<enabled>true</enabled>");
+        pw.println(INDENT.repeat(3) + "<level>MEDIUM</level>");
+        pw.println(INDENT.repeat(2) + "</p" + plugins[i].getId() + ">");
+      }
+    } catch (e) {
+      print(e);
```

Comment:
Would be good if this failed the build, otherwise we are going to miss these errors, though hopefully there will be none.